nuxeo-runtime/nuxeo-runtime-tomcat-adapter/src/main/java/org/nuxeo/runtime/tomcat/NuxeoDeployer.java

```diff
@@ -50,54 +54,70 @@ public String getHome() {
-                    System.out.println("Deployer and/or common JAR (nuxeo-runtime-deploy* | nuxeo-common*) not found in "
-                            + bundles);
-                    return;
-                }
-                ArrayList<URL> urls = new ArrayList<URL>();
-                File[] files = lib.listFiles();
-                if (files != null) {
-                    for (File f : files) {
-                        if (f.getPath().endsWith(".jar")) {
-                            urls.add(f.toURI().toURL());
-                        }
+            File homeDir = resolveHomeDirectory();
+            File bundles = new File(homeDir, "bundles");
+            File lib = new File(homeDir, "lib");
+            File deployerJar = FrameworkBootstrap.findFileStartingWidth(bundles, "nuxeo-runtime-deploy");
+            File commonJar = FrameworkBootstrap.findFileStartingWidth(bundles, "nuxeo-common");
+            if (deployerJar == null || commonJar == null) {
+                System.out.println("Deployer and/or common JAR (nuxeo-runtime-deploy* | nuxeo-common*) not found in "
+                        + bundles);
+                return;
+            }
+            ArrayList<URL> urls = new ArrayList<URL>();
+            File[] files = lib.listFiles();
+            if (files != null) {
+                for (File f : files) {
+                    if (f.getPath().endsWith(".jar")) {
+                        urls.add(f.toURI().toURL());
                     }
                 }
-                files = bundles.listFiles();
-                if (files != null) {
-                    for (File f : files) {
-                        if (f.getPath().endsWith(".jar")) {
-                            urls.add(f.toURI().toURL());
-                        }
+            }
+            files = bundles.listFiles();
+            if (files != null) {
+                for (File f : files) {
+                    if (f.getPath().endsWith(".jar")) {
+                        urls.add(f.toURI().toURL());
                     }
                 }
-                urls.add(homeDir.toURI().toURL());
-                urls.add(new File(homeDir, "config").toURI().toURL());
-                try (URLClassLoader cl = new URLClassLoader(urls.toArray(new URL[urls.size()]), parentCl)) {
-                    System.out.println("# Running Nuxeo Preprocessor ...");
-                    Class<?> klass = cl.loadClass("org.nuxeo.runtime.deployment.preprocessor.DeploymentPreprocessor");
-                    Method main = klass.getMethod("main", String[].class);
-                    main.invoke(null, new Object[] { new String[] { homeDir.getAbsolutePath() } });
-                    System.out.println("# Preprocessing done.");
-                }
+            }
+            urls.add(homeDir.toURI().toURL());
+            urls.add(new File(homeDir, "config").toURI().toURL());
+            try (URLClassLoader cl = new URLClassLoader(urls.toArray(new URL[urls.size()]), parentCl)) {
```

Comment:
Can you change `urls.size()` to 0 ? It has better performance https://shipilev.net/blog/2016/arrays-wisdom-ancients/